### DISCUSSION TITLE
webstorm 总是不认识 vue 的方法和模板语法咋办

### DISCUSSION THREAD

**Discussion by Arondight** [Arondight/Adachi-BOT#785] [Q&A]
![图片](https://user-images.githubusercontent.com/13673702/161762591-15364ed0-d47f-469c-9942-c5937c6124cb.png)

![图片](https://user-images.githubusercontent.com/13673702/161762636-672c4694-3481-4ec0-9417-0cfeaca88ef9.png)

@Mark9804

**Comment by mark9804** [comment]
看一下插件里面有没有 `Vue.js` ，一般会有

看一下在编辑的文件是不是 `.vue` 结尾的

**Reply by Arondight** [reply]
html 结尾的

**Reply by Arondight** [reply]
我研究一下吧，好像改后缀也有问题

**Reply by mark9804** [reply]
webstorm 靠 vue.js 插件来支持 vue，在 .html 结尾的文件里面这个插件就摸了，这点确实挺蠢的，我也顶着警告写

目前我知道的解决方法只有把 vue 的代码全部扔到 .vue 文件里面，或者就配置忽略警告

**Reply by Arondight** [reply]
把 html 文件类型改成 vue 这些问题依然存在，而且没法快捷打开浏览器了，必须得本地起一个 server 才行，感觉很蠢

**Reply by mark9804** [reply]
> 把 html 文件类型改成 vue

别这样，把 vue 组件独立出来，和 html 分开，就可以了，webstorm 是蠢但不能用更蠢的方式来解决问题啊

**Reply by Arondight** [reply]
> > 把 html 文件类型改成 vue
> 
> 别这样，把 vue 组件独立出来，和 html 分开，就可以了，webstorm 是蠢但不能用更蠢的方式来解决问题啊

前一阵摸鱼学 vue 的时候发现 webstorm 不认识 vite

**Reply by mark9804** [reply]
webstorm 是用插件的方式来支持 vue 的, vite 有一个第三方插件我没试过

[Vite Integrated - IntelliJ IDEs Plugin | Marketplace](https://plugins.jetbrains.com/plugin/16897-vite-integrated/)

**Reply by mark9804** [reply]
> 前一阵摸鱼学 vue 的时候发现 webstorm 不认识 vite

支持了

<img width="428" alt="image" src="https://user-images.githubusercontent.com/9006264/168407456-c168a65e-0ba1-46b1-b34d-45d5f6de633d.png">

**Reply by Arondight** [reply]
> 持了

更新了，还是没能给我解析这个 [vite.config.js](https://github.com/Arondight/express-auth-demo/blob/664e26fe9499558bbcbcc5bdeddba6aa37f22df0/vite.config.js)，项目中的 import 也不认识，自然也没法分析和跳转，感觉还是差了点

**Reply by mark9804** [reply]
我刚刚试了一下都是可以的啊
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/9006264/168411704-61ad72e6-4d65-4f28-a407-541e8574678d.png">

**Reply by mark9804** [reply]
试了一下老大哥的，感觉没问题
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/9006264/168412007-d9703dc2-2e49-4ef1-ba79-86e8307f772e.png">

**Reply by Arondight** [reply]
vue 里面的 import 别名不认识

**Reply by mark9804** [reply]
确实不认得，还是蠢了点

**Reply by mark9804** [reply]
> vue 里面的 import 别名不认识

设置 → 语言和框架 → JavaScript → Webpack → 手动，然后把配置文件指向 vite.config.js

<img width="723" alt="image" src="https://user-images.githubusercontent.com/9006264/168414410-697028a6-64d1-4ac6-a00a-e038a77af10c.png">

**Reply by Arondight** [reply]
🐮，晚上试试

**Reply by mark9804** [reply]
> vue 里面的 import 别名不认识

webstorm 现在原生支持 vite 了

<img width="235" alt="image" src="https://user-images.githubusercontent.com/9006264/194760610-3c906c7c-dfb1-4f09-b39e-f5b51ac95988.png">
